calcos/calcos.py

```diff
@@ -2580,6 +2581,136 @@ def printSwitchMessages(self, messages, input):
             for msg in messages:
                 cosutil.printMsg(msg, VERBOSE)
 
+    def CheckforAddSplitWavecal(self, association, wavecal_info, debug=False):
+        """A split wavecal entry needs to be inserted if these conditions are met
+
+        Parameters:
+        -----------
+
+        Returns:
+        --------
+
+        boolean
+            True if split wavecal is to be inserted
+        """
+        info = self.info
+        reffiles = self.reffiles
+
+        if info['detector'] == 'NUV':
+            if debug:
+                cosutil.printMsg("Don't add split wavecal because observation is NUV")
+            return False
+
+        if 'EXP-SWAVE' not in association.asn_info['memtype']:
+            if debug:
+                cosutil.printMsg("Don't add split wavecal because association has no exp_swave members")
+            return False
+
+        # If the other segment exists in this association and has the MustAddSplitWavecal flag set,
+        # set it for this obs
+        othersegmentobs = self.getOtherSegmentObs(association)
+        if othersegmentobs is not None:
+            othersegmentinfo = othersegmentobs.info
+            if othersegmentinfo['addsplitwavecal']:
+                if debug:
+                    cosutil.printMsg("Add split wavecal because other segment is")
```

Comment:
@stscirij I think this message isn't complete?

Reply:
Actually that's close to what I was planning to say, which is more like "Add split wavecal because other segment has one"

Reply:
Got it! Makes sense now that I read it not in anticipation of what comes next.